sonar-plugin/sonar-javascript-plugin/src/main/java/org/sonar/plugins/javascript/analysis/AbstractAnalysis.java

```diff
@@ -110,7 +113,10 @@ protected void analyzeFile(InputFile file, @Nullable List<String> tsConfigs, @Nu
           CacheAnalysis.fromResponse(response.ucfgPaths(), response.cpdTokens()),
           file
         );
-        consumers.accept(new JsFile(file));
+        Node responseAst = response.ast();
+        if (responseAst != null) {
+          consumers.accept(new JsFile(file, ESTreeFactory.from(responseAst, ESTree.Program.class)));
+        }
```

Comment:
should we comment here a few examples of cases where this happens? like that
```suggestion
        Node responseAst = response.ast();
        if (responseAst != null) {
          // When we haven't serialized the AST:
          // either because no consumer is listening
          // or the file extension or AST nodes are unsupported
          consumers.accept(new JsFile(file, ESTreeFactory.from(responseAst, ESTree.Program.class)));
        }
```